.github/workflows/wasm.yml

```diff
@@ -0,0 +1,62 @@
+name: "rebuild wasm"
+
+on:
+  workflow_dispatch:
+
+jobs:
+  rebuild-wasm:
+    runs-on: ubuntu-latest
+    steps:
+        - uses: actions/checkout@v2
+        - name: install dependencies
+          run: |-
+            curl -fsSL https://deno.land/x/install/install.sh | sh
+            echo "$HOME/.deno/bin" >> $GITHUB_PATH
+
+            rustup target add wasm32-unknown-unknown
+
+            # This must match the version in hash/_wasm/Cargo.lock:
+            cargo install -f wasm-bindgen-cli --version 0.2.74
+
+        - name: rebuild generated files
+          id: build
+          run: |-
+            ./hash/_wasm/build.ts
+
+            declare had_changes="$(
+              [[ -n "$(git status --porcelain)" ]] && echo true || echo false
+            )"
+            echo "::set-output name=had-changes::${had_changes}"
+
+        - name: commit and push generated files
+          if: success() && steps.build.outputs.had-changes == 'true'
+          run: |
+            declare br="
+            "
+            declare branch="$(echo ${GITHUB_REF#refs/heads/})"
+            declare date="$(git log -1 --pretty=format:'%ad')"
+            declare short_hash="$(git rev-parse --short=8 HEAD)"
+            declare short_subject="$(git log -1 --pretty=format:'%s')"
+            declare short_subject="$(git log -1 --pretty=format:'%s')"
+            if [[ ${#short_subject} -gt 50 ]]; then
+              short_subject="${short_subject:0:49}…"
+            fi
+            declare author_name="$(git log -1 --pretty=format:'%an')"
+            declare author_email="$(git log -1 --pretty=format:'%ae')"
+            declare current_action_url="https://github.com/${GITHUB_REPOSITORY}/actions/runs/${{ github.run_id }}"
+
+            export GIT_AUTHOR_DATE="${date}"
+            export GIT_AUTHOR_NAME="${author_name}"
+            export GIT_AUTHOR_EMAIL="${author_email}"
+
+            export GIT_COMMITTER_DATE="${date}"
+            export GIT_COMMITTER_NAME="github-actions"
+            export GIT_COMMITTER_EMAIL="41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
Not super familiar with using github actions to commit changes to the PR -- what is this email address and why does it work?

Reply:
This is just a cosmetic choice and can be dropped if you prefer. It's the email address of the internal GitHub Actions bot account, so by specifying it here the commit gets a GitHub actions logo as its committer, making it stand out more clearly as being generated.